src/test/scala/org/broadinstitute/dsde/firecloud/mock/MockMethodsServer.scala

```diff
@@ -16,13 +16,165 @@ import DefaultJsonProtocol._
 
 object MockMethodsServer {
 
+
+
   val methodsServerPort = 8989
 
   val methodsUrl = FireCloudConfig.Agora.authPrefix + FireCloudConfig.Agora.methodsPath
   val configsUrl = FireCloudConfig.Agora.authPrefix + FireCloudConfig.Agora.configurationsPath
 
+
+
+  val allPermsConfPath1="/namespace/mockAgoraOwnerAllFivePermission/1/permissions"
+  val allPermsConfPath2="/namespace/mockAgoraOwnerAllPermission/1/permissions"
+  val readerPermsConfPath="/namespace/mockAgoraReader/1/permissions"
+  val noPermsNothingConfPath="/namespace/mockAgoraNoPermsNothing/1/permissions"
+  val noPermsEmptyConfPath="/namespace/mockAgoraNoPermsEmpty/1/permissions"
+
+
   /****** Mock Data ******/
 
+  def generateRandomFireCloudRole():String = {
+    val randomNum=scala.util.Random.nextInt(2)
+    if(randomNum==0)
+      {
+        return "NO ACCESS"
+      }
+    else if(randomNum==1)
+      {
+        return "READER"
+      }
+    else
+      {
+        return "OWNER"
+      }
+
+  }
+
+
+  def generateRandomFireCloudPermissions():List[FireCloudPermission] ={
+    List.tabulate(randomPositiveInt()-1)(
+      n=>
+        FireCloudPermission(
+          user=Some(randomAlpha()),
+          role=Some(generateRandomFireCloudRole())
+        )
+    )
+  }
+
+
+
+
+
+  def generateRandomAgoraRoles():List[String] =
+  {
+    //each of the 5 individual roles
+    //has a 1/2 probability of showing up
+    val randomNum=scala.util.Random.nextInt(2)
+    //println("A randomNum is "+randomNum)
+    var aList=List[String]()
+    if(randomNum ==0) {
+      aList=(aList :+ "Read")
+    }
+    val randomNum2=scala.util.Random.nextInt(2)
+    if(randomNum2 ==0 )
+    {
+      aList=(aList :+ "Write")
+    }
+    val randomNum3=scala.util.Random.nextInt(2)
+    if(randomNum3  ==0)
+    {
+      aList=(aList :+ "Create")
+    }
+    val randomNum4=scala.util.Random.nextInt(2)
+    if(randomNum4 ==0)
+    {
+      aList=(aList :+ "Redact")
+    }
+    val randomNum5=scala.util.Random.nextInt(2)
+    if(randomNum5 ==0)
+    {
+      aList=(aList :+ "Manage")
+    }
+    return aList
+  }
+
+
+
+
+
+
+ val mockAgoraPermissions: List[AgoraPermission] = {
+    List.tabulate(randomPositiveInt()-1)(
+    n=>
+      AgoraPermission(
+        user=Some(randomAlpha()),
+        roles=Some(generateRandomAgoraRoles())
+      )
+    )
+  }
+
+  val mockAgoraOwnerAllPermission:List[AgoraPermission] ={
+    var aList=List[String]()
+    aList=(aList:+"All")
+    var anAgoraPermission=AgoraPermission(
+      user=Some("an_owner"),
+      roles=Option(aList))
+    var aPermList=List[AgoraPermission]()
+    aPermList=(aPermList:+anAgoraPermission)
+    aPermList
+  }
+
+  val mockAgoraOwnerAllFivePermission:List[AgoraPermission] ={
+    var aList=List[String]()
+    aList=(aList:+"Read")
+    aList=(aList:+"Write")
+    aList=(aList:+"Create")
+    aList=(aList:+"Redact")
+    aList=(aList:+"Manage")
```

Comment:
The above is a very strong anti-pattern. Just create a list:
```
val agoraRoles = List("Read", "Write", "Create", "Redact", "Manage")
```
And then you can use that in the roles assignment below.